ctrd/image_commit.go

```diff
@@ -0,0 +1,320 @@
+package ctrd
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"runtime"
+	"strings"
+	"time"
+
+	"github.com/alibaba/pouch/apis/types"
+	"github.com/alibaba/pouch/pkg/randomid"
+
+	"github.com/containerd/containerd"
+	"github.com/containerd/containerd/content"
+	"github.com/containerd/containerd/diff"
+	"github.com/containerd/containerd/errdefs"
+	"github.com/containerd/containerd/images"
+	"github.com/containerd/containerd/platforms"
+	"github.com/containerd/containerd/rootfs"
+	"github.com/containerd/containerd/snapshots"
+	digest "github.com/opencontainers/go-digest"
+	"github.com/opencontainers/image-spec/identity"
+	specs "github.com/opencontainers/image-spec/specs-go"
+	ocispec "github.com/opencontainers/image-spec/specs-go/v1"
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+)
+
+var (
+	emptyGZLayer           = digest.Digest("sha256:4f4fb700ef54461cfa02571ae0db9a0dc1e0cdb5577484a6d75e68dc38e8acc1")
+	containerdUncompressed = "containerd.io/uncompressed"
+	manifestType           = images.MediaTypeDockerSchema2Manifest
+	configType             = images.MediaTypeDockerSchema2Config
+	layerType              = images.MediaTypeDockerSchema2LayerGzip
+)
+
+// CommitConfig defines options for committing a container image
+type CommitConfig struct {
```

Comment:
I think this part we can move to `api/`, create a new directory like `api/backend`

Reply:
just keep it here because it maybe cause the cycle dependency. API -> Daemon -> ctrd -> API... maybe